### PR TITLE
Craig 911: NoVpcTile

### PR DESCRIPTION
**Pull request by terechc**
Moved NoVpcTile from CRAIG to here so that the header still appears when the tile is shown

### CODE REVIEW

File: src/components/crud-form-pages/RoutingTables.js [hunk 1]
```diff
@@ -40,6 +41,11 @@ export const RoutingTables = (props) => {
         submissionFieldName: "routing_tables",
         hideName: true,
       }}
+      overrideTile={
+        props.craig.store.json.vpcs.length === 0
```
Comment (jvallexm) [inline src/components/crud-form-pages/RoutingTables.js:45]: use `props.vpcList` here

File: src/components/crud-form-pages/SecurityGroups.js [hunk 2]
```diff
@@ -41,6 +42,11 @@ export const SecurityGroups = (props) => {
         hide: true,
         hideName: true,
       }}
+      overrideTile={
+        props.craig.store.json.vpcs.length === 0
```
Comment (jvallexm) [inline src/components/crud-form-pages/SecurityGroups.js:46]: use `props.vpcList` here

File: src/components/crud-form-pages/Vpc.js [hunk 3]
```diff
@@ -4,6 +4,23 @@ import IcseFormTemplate from "../IcseFormTemplate";
 import PropTypes from "prop-types";
 import VpcNetworkForm from "../forms/VpcForm";
 
+import { CloudAlerting } from "@carbon/icons-react";
+import { Tile } from "@carbon/react";
+import "./no-secrets-manager-tile.css";
+
+export const NoVpcTile = (resource) => {
```
Comment (jvallexm) [inline src/components/crud-form-pages/Vpc.js:11]: move no vpc tile to own file

File: src/components/crud-form-pages/VpnGateways.js [hunk 4]
```diff
@@ -33,6 +34,11 @@ export const VpnGateways = (props) => {
         hide: true,
         hideName: true,
       }}
+      overrideTile={
+        props.craig.store.json.vpcs.length === 0
```
Comment (jvallexm) [inline src/components/crud-form-pages/VpnGateways.js:38]: use `props.vpcList` here

File: src/components/crud-form-pages/VpnServers.js [hunk 5]
```diff
@@ -37,6 +38,8 @@ export const VpnServers = (props) => {
       overrideTile={
         props.noSecretsManager ? (
           <NoSecretsManagerTile text="to enable VPN Servers." />
+        ) : props.craig.store.json.vpcs.length === 0 ? (
+          NoVpcTile("Routing Tables")
```
Comment (jvallexm) [inline src/components/crud-form-pages/VpnServers.js:42]: let's render noVpcTile first

File: src/components/crud-form-pages/VpnServers.js [hunk 6]
```diff
@@ -37,6 +38,8 @@ export const VpnServers = (props) => {
       overrideTile={
         props.noSecretsManager ? (
           <NoSecretsManagerTile text="to enable VPN Servers." />
+        ) : props.craig.store.json.vpcs.length === 0 ? (
```
Comment (jvallexm) [inline src/components/crud-form-pages/VpnServers.js:41]: use `props.vpcList` here

File: src/components/crud-form-pages/Vsi.js [hunk 7]
```diff
@@ -50,6 +51,11 @@ export const Vsi = (props) => {
         hide: true,
         hideName: true,
       }}
+      overrideTile={
+        props.craig.store.json.vpcs.length === 0
```
Comment (jvallexm) [inline src/components/crud-form-pages/Vsi.js:55]: use `props.vpcList` here

File: src/components/crud-form-pages/VsiLoadBalancer.js [hunk 8]
```diff
@@ -34,6 +35,11 @@ export const VsiLoadBalancer = (props) => {
         hide: true,
         hideName: true,
       }}
+      overrideTile={
+        props.craig.store.json.vpcs.length === 0
```
Comment (jvallexm) [inline src/components/crud-form-pages/VsiLoadBalancer.js:39]: use `props.vpcList` here

File: src/components/crud-form-pages/VpnServers.js (hunk 5 above)
Comment (jvallexm) [reply src/components/crud-form-pages/VpnServers.js:42]: also should be vpn server

**Review by jvallexm — APPROVED** [review]
lgtm, please get review from evelyn. you can probably just post images of this working in CRAIG in the issue and tag her

File: src/components/crud-form-pages/NoVpcTile.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { CloudAlerting } from "@carbon/icons-react";
+import { Tile } from "@carbon/react";
+import "./no-secrets-manager-tile.css";
+
+export const NoVpcTile = (resource) => {
+  return (
+    <Tile className="tileBackground displayFlex alignItemsCenter wrap marginTop">
+      <CloudAlerting size="24" className="iconMargin" /> No VPCs have
+      been created. Create one from the
```
Comment (evpirnia) [inline src/components/crud-form-pages/NoVpcTile.js:10]: Can we copy the message on the Power Page? 

So something like "No VPCs have been created. Return to the Virtual Private Clouds Page."

@jvallexm we also need to add an issue to keep the headers on the Power pages even if the message is shown.
Screenshot 2023-10-17 at 12 19 30 PM

**Review by evpirnia — CHANGES_REQUESTED** [review]

File: src/components/crud-form-pages/NoVpcTile.js (hunk 9 above)
Comment (jvallexm) [reply src/components/crud-form-pages/NoVpcTile.js:10]: yeah we need a new issue for that

Comment (terechc) [reply src/components/crud-form-pages/NoVpcTile.js:10]: @evpirnia Going to say ```"No VPCs have been created. Return to the Virtual Private Clouds Page to enable {resource}."``` Is that what you mean?

Comment (evpirnia) [reply src/components/crud-form-pages/NoVpcTile.js]: sure!

**Pull request merged**